File: src/main/java/com/robyrodriguez/stackbuster/types/BadgeType.java
package com.robyrodriguez.stackbuster.types;

/**
 * Badge types
 */
public enum BadgeType {
    /** 1k views */
    POPULAR_QUESTION(1000),
    /** 2.5k views */
    NOTABLE_QUESTION(2500),
    /** 10k views */
    FAMOUS_QUESTION(10000),
    /** 25 shared clicks */
    ANNOUNCER(25),
    /** 300 shared clicks */
    BOOSTER(300),
    /** 1k shared clicks */
    PUBLICIST(1000);

    private int clicks;

    BadgeType(int clicks) {
        this.clicks = clicks;
    }

    public int getClicks() {
        return clicks;
    }

    public boolean isUserInvolved() {
        return this == ANNOUNCER || this == BOOSTER || this == PUBLICIST;
    }
}

File: functions/functions/types/badge.type.js
const BadgeType = {
    POPULAR_QUESTION: 'POPULAR_QUESTION',
    NOTABLE_QUESTION: 'NOTABLE_QUESTION',
    FAMOUS_QUESTION: 'FAMOUS_QUESTION',

    ANNOUNCER: 'ANNOUNCER',
    BOOSTER: 'BOOSTER',
    PUBLICIST: 'PUBLICIST',

    isUserInvolved: function (type) {
        return type === this.ANNOUNCER || type === this.BOOSTER || type === this.PUBLICIST
    },
    isValid: function (type) {
        for (const key of Object.keys(this)) {
            if (typeof this[key] !== 'function' && this[key] === type)
                return true
        }
    }
}

module.exports = BadgeType;
